frontend/src/Components/Game.jsx

```diff
@@ -0,0 +1,121 @@
+import React from "react";
+import Map from "./Map";
+import "./Game.css";
+import Player from "./Player";
+import { tileNames, tilesMap } from "./tilesMap.js";
+
+class Game extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      x: 3,
+      y: 4,
+      keyCode: 40,
+      canMoove: true
+    };
+  }
+
+  // Fonction qui écoute les input du clavier sur toutela fenetre
+
+  componentDidMount() {
+    window.onkeydown = event => {
+      if (this.state.canMoove) {
+        this.setState({ canMoove: false });
+        setTimeout(() => {
+          this.setState({ canMoove: true });
+        }, 120);
+        this.getMouvment(event);
+      }
+    };
+  }
+
+  // Fonction qui récupere l'input du clavier pour afficher le bon asset et faire bouger le player
+  getMouvment(event) {
+    let newKeyCode = event.keyCode;
+    switch (newKeyCode) {
+      case 37:
+        event.preventDefault();
+        let newPosition = this.state.x - 1;
+        if (
+          newPosition < 0 ||
+          tilesMap[this.state.y][this.state.x - 1].includes("Z")
+        ) {
+          break;
+        } else {
+          this.setState({
+            x: newPosition,
+            keyCode: newKeyCode
+          });
+        }
+        break;
+
+      case 38:
+        event.preventDefault();
+        let newPosition2 = this.state.y - 1;
+        if (
+          newPosition2 < 0 ||
+          tilesMap[this.state.y - 1][this.state.x].includes("Z")
+        ) {
+          break;
+        } else {
+          this.setState({
+            y: newPosition2,
+            keyCode: newKeyCode
+          });
+        }
+        break;
+
+      case 39:
+        event.preventDefault();
+        let newPosition3 = this.state.x + 1;
+        if (
+          newPosition3 > 19 ||
+          tilesMap[this.state.y][this.state.x + 1].includes("Z")
+        ) {
+          break;
+        } else {
+          this.setState({
+            x: newPosition3,
+            keyCode: newKeyCode
+          });
+        }
+        break;
+
+      case 40:
```

Comment:
As long as we're speaking about weird conditions: what does "40" mean ? You should create a named constant ("LEFT" ?) in a `conf.js` file and use it.
Bonus point: it would make key customisation easier on the long run ;-)

Reply:
Constants renamed